src/Rules/SetValueMethodRule.php

```diff
@@ -89,6 +89,10 @@ public function processNode(Node $node, Scope $scope): array
 		$property = $class->getProperty($fieldName, $scope);
 		$propertyType = $property->getWritableType();
 
+		if (!$property->isWritable()) {
```

Comment:
This doesn't seem right - if case the property is not writable, we need to resolve the writeable error first, instead it will result in the wrong error -> readable type doesn't match the type being assigned/written.

Reply:
I am not sure what you mean.

- Without https://github.com/phpstan/phpstan-src/pull/2294, `$property->getWritableType() === $property->getReadableType()` so this would be a no-op.
- And with it, this is necessary since `$property->getWritableType()` will be a bottom type `void` when the property is not writeable. And it will always be readable or writeable when it exists.

What does not seem right to me is writing to read-only properties but you have created `setReadOnlyValue()` so :woman_shrugging:

Reply:
Yes, and since readable type will be different, this PHPStan will improperly throw an error that readable type don't accept the value.

Reply:
I do not think it will. PHPStan is not aware that `setReadOnlyValue()` or `setValue()` modifies the property. Only this rule ensures it.